### PR TITLE
7.x.x/hd/fix motionblur alpha

### PR DESCRIPTION
**Pull request by H3idi-X**
# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

Motionblur.compute doesn't work well when both Color buffer and Postprocessing buffer format are R16G16B16A16.
It makes the gameview image lighter in playmode.
https://fogbugz.unity3d.com/f/cases/1333592/

This PR can fix above.
---
### Testing status
Describe what manual/automated tests were performed for this PR

I did manual test and compared to the result with other HDRP version.
---
### Comments to reviewers
Notes for the reviewers you have assigned.

We discussed the issue and this PR in the following thread:
https://unity.slack.com/archives/C6Y79CZM0/p1620107244387900
https://unity.slack.com/archives/C6Y79CZM0/p1620359510462500

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute [hunk 1]
```diff
@@ -127,7 +128,7 @@ void MirrorWeights(float depth1, float depth2, float motionVec1, float motionVec
 // Sample processing 
 // ---------------------------------------------
 
-float3 ProcessSampleFastPath(uint sampleNumber, float2 dir, float invSampleCount, float2 centralUV, float randomVal, float dirSign)
+float3 ProcessSampleFastPathF3(uint sampleNumber, float2 dir, float invSampleCount, float2 centralUV, float randomVal, float dirSign)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute:131]: you can do is more simply

#pragma kernel MotionBlurF3CS 
#pragma kernel MotionBlurF4CS ENABLE_ALPHA

#if ENABLE_ALPHA
#define ProcessSampleFastPath ProcessSampleFastPathF4
#else
#define ProcessSampleFastPath ProcessSampleFastPathF3
#endif

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute:131]: This avoid to duplicate all this code

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute [hunk 2]
```diff
@@ -162,13 +175,53 @@ float4 ProcessSample(uint sampleNumber, float2 dir, float invSampleCount, float2
     return float4(sampleColor, sampleWeight);
 }
 
+float4 ProcessSampleFastPath(uint sampleNumber, float2 dir, float invSampleCount, float2 centralUV, float randomVal, float dirSign)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute:178]: ProcessSampleFastPathF4 is exactly same than ProcessSampleFastPath, don't think it is expect?

Comment (higeometry) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute:178]: My bad. you are right.

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute (hunk 1 above)
Comment (higeometry) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute:131]: You are right.
How do you think about this fix?

As far as I've tested it is working.

```
#include "Packages/com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlurCommon.hlsl"

#pragma kernel MotionBlurF3CS
#pragma kernel MotionBlurF4CS ENABLE_ALPHA

#if ENABLE_ALPHA
#define CTYPE float4
#define CTYPE_SWIZZLE xyzw
#else
#define CTYPE float3
#define CTYPE_SWIZZLE xyz
#endif
//#pragma enable_d3d11_debug_symbols

RW_TEXTURE2D_X(CTYPE, _OutputTexture);


void KernelMain(uint3 dispatchThreadId ) // 
{
    UNITY_XR_ASSIGN_VIEW_INDEX(dispatchThreadId.z);
    CTYPE debugColor = 1.0f;
    // Get central sample early.
    CTYPE centralColor = _InputTexture[COORD_TEXTURE2D_X(dispatchThreadId.xy)].CTYPE_SWIZZLE;
     .
     .
     .
     .
     .
}
// ---------------------------------------------
// Kernel
// ---------------------------------------------
[numthreads(16, 16, 1)]
void MotionBlurF4CS(uint3 dispatchThreadId : SV_DispatchThreadID)
{
    KernelMain(dispatchThreadId);
}
[numthreads(16, 16, 1)]
void MotionBlurF3CS(uint3 dispatchThreadId : SV_DispatchThreadID)
{
    KernelMain(dispatchThreadId);
}

```

Comment (higeometry) [reply com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/MotionBlur.compute]: I committed this and started yamato test.

**Review by sebastienlagarde — APPROVED** [review]
Thanks for the refactor, way better!

**Comment by higeometry** [comment]
I ran required test, _ABV for SRP repository - fast 2019.4, and I believe all checks have passed but GitHub says "Merging is blocked". Can I do something for this?

**Comment by sebastienlagarde** [comment]
Merged for you, thanks

**Pull request merged**